### PR TITLE
fix(fallback-integrity): insure integrity is on module config

### PR DESCRIPTION
**Pull request by JAdshead**
#### _Describe your changes in detail_
Fallbacks integrity not being consistently included in module config for following reasons:
- Production bundler would override `bundle.integrity.manifest.json` removing previously created integrity value for external fallbacks.
- Race condition where `bundle.integrity.manifest` would be updated after being read with different fallback integrity values.

This change fixes the race condition where multiple required externals could override values in `bundle.integrity.manifest`, includes small refactor for updating module config, also ensures module configs required externals is handled the same for both bundlers.
This change also creates integrity values for both node and browser builds.

#### _Why is this change required? What issue does it resolve?_

Ensure integrity values are included in module config for external fallbacks.

#### _How does your change affect the rest of the code._ 

* Test suite
* tested packed deps with locally created module which requires externals. Validated that `module-config.json` always includes integrity values. Successfully loaded modules and fallbacks built with both bundlers. 

#### Check boxes that apply:
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/one-app-dev-bundler/utils/bundle-external-fallbacks.js [hunk 1]
```diff
@@ -69,6 +71,27 @@ export const bundleExternalFallbacks = async () => {
...
+      })?.packageJson.version;
+      const semanticRange = packageJson.dependencies[externalName];
+      return {
+        ...obj,
+        [externalName]: {
+          name: externalName,
+          version,
+          semanticRange,
+          integrity: integrityManifest[`${externalName}.browser`], // prevent breaking change
+          browserIntegrity: integrityManifest[`${externalName}.browser`],
```
Comment (JAdshead) [inline packages/one-app-dev-bundler/utils/bundle-external-fallbacks.js:88]: I am not 100% on if we need `browserIntegrity` and `integrity`

File: packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js [hunk 2]
```diff
@@ -18,17 +18,18 @@ import ssri from 'ssri';
...
-    integrityObject = JSON.parse(await fs.promises.readFile(fileName, 'utf8'));
+    integrityObject = JSON.parse(fs.readFileSync(fileName, 'utf8'));
   } catch (e) {
     // empty catch in case file does not exist
   }
 
   integrityObject[bundleName] = integrityString;
-
-  return fs.promises.writeFile(fileName, JSON.stringify(integrityObject, null, 2));
+  return fs.writeFileSync(fileName, JSON.stringify(integrityObject, null, 2));
```
Comment (Matthew-Mallimo) [inline packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js:32]: Can all of this be put in the try-catch? If we fail to read the file, do we still want to be writing an empty object?

Comment (Matthew-Mallimo) [inline packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js:32]: I dont think we should be pretty formatting this, its just wasted bytes

File: packages/one-app-dev-bundler/utils/bundle-external-fallbacks.js (hunk 1 above)
Comment (Matthew-Mallimo) [reply packages/one-app-dev-bundler/utils/bundle-external-fallbacks.js:88]: is `browserIntegrity` used anywhere already, or is that new to this PR? I don't think it makes sense to maintain 2 of the same values, even though it is a bit more clear

Comment (JAdshead) [reply packages/one-app-dev-bundler/utils/bundle-external-fallbacks.js:88]: its not used anywhere, but integrity is. Id be more inclined to remove integrity, especially as i don't think production builds currently have integrity included

File: packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js (hunk 2 above)
Comment (JAdshead) [reply packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js:32]: if it fails this the error should bubble up and fail the build. The try catch squashes it

Comment (JAdshead) [reply packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js:32]: This file should be named in `.gitignore` and can be cleaned up between builds, I personally value the readability.

Comment (Matthew-Mallimo) [reply packages/one-app-dev-bundler/esbuild/plugins/generate-integrity-manifest.js:32]: Ah I see now. I was thinking this was the module config file.

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by dogpatch626 — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**